### ISSUE TITLE
deprecated constants warning which will be removed in HA Core 2025.1

### ISSUE THREAD

**Issue by scooper1** [AlexxIT/SonoffLAN#1278]
deprecated constants warning which will be removed in HA Core 2025.1.
first occurs in beta 2024.1

```
2023-12-28 02:49:50.412 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_AUTO was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.AUTO instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.422 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_COOL was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.COOL instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.426 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_DRY was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.DRY instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.431 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_HEAT was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.HEAT instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.436 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_HEAT_COOL was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.HEAT_COOL instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.441 WARNING (MainThread) [homeassistant.components.climate.const] HVAC_MODE_OFF was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.OFF instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.445 WARNING (MainThread) [homeassistant.components.climate.const] SUPPORT_PRESET_MODE was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.PRESET_MODE instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.448 WARNING (MainThread) [homeassistant.components.climate.const] SUPPORT_TARGET_TEMPERATURE was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.TARGET_TEMPERATURE instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.453 WARNING (MainThread) [homeassistant.components.climate.const] SUPPORT_TARGET_TEMPERATURE_RANGE was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.TARGET_TEMPERATURE_RANGE instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.458 WARNING (MainThread) [homeassistant.const] TEMP_CELSIUS was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfTemperature.CELSIUS instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.464 WARNING (MainThread) [homeassistant.components.fan] SUPPORT_PRESET_MODE was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use FanEntityFeature.PRESET_MODE instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.468 WARNING (MainThread) [homeassistant.components.fan] SUPPORT_SET_SPEED was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use FanEntityFeature.SET_SPEED instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.471 WARNING (MainThread) [homeassistant.components.fan] SUPPORT_PRESET_MODE was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use FanEntityFeature.PRESET_MODE instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.475 WARNING (MainThread) [homeassistant.components.fan] SUPPORT_SET_SPEED was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use FanEntityFeature.SET_SPEED instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.484 WARNING (MainThread) [homeassistant.const] ELECTRIC_CURRENT_AMPERE was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfElectricCurrent.AMPERE instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.488 WARNING (MainThread) [homeassistant.const] ELECTRIC_POTENTIAL_VOLT was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfElectricPotential.VOLT instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.492 WARNING (MainThread) [homeassistant.const] ENERGY_KILO_WATT_HOUR was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.496 WARNING (MainThread) [homeassistant.const] POWER_WATT was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfPower.WATT instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
2023-12-28 02:49:50.501 WARNING (MainThread) [homeassistant.const] TEMP_CELSIUS was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfTemperature.CELSIUS instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues

```

**Comment by yousaf465** [comment]
```
Logger: homeassistant.const
Source: helpers/deprecation.py:204
First occurred: 10:02:05 (22 occurrences)
Last logged: 10:02:31

POWER_WATT was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfPower.WATT instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
TEMP_CELSIUS was used from ble_monitor, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfTemperature.CELSIUS instead, please create a bug report at https://github.com/custom-components/ble_monitor/issues
TEMP_FAHRENHEIT was used from ble_monitor, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfTemperature.FAHRENHEIT instead, please create a bug report at https://github.com/custom-components/ble_monitor/issues
ENERGY_KILO_WATT_HOUR was used from huawei_solar, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/wlcrs/huawei_solar/issues
POWER_WATT was used from huawei_solar, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfPower.WATT instead, please create a bug report at https://github.com/wlcrs/huawei_solar/issues
```

```
Logger: homeassistant.components.climate.const
Source: helpers/deprecation.py:204
Integration: Climate (documentation, issues)
First occurred: 10:02:05 (17 occurrences)
Last logged: 10:02:20

HVAC_MODE_HEAT_COOL was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.HEAT_COOL instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
HVAC_MODE_OFF was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use HVACMode.OFF instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
SUPPORT_PRESET_MODE was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.PRESET_MODE instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
SUPPORT_TARGET_TEMPERATURE was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.TARGET_TEMPERATURE instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues
SUPPORT_TARGET_TEMPERATURE_RANGE was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use ClimateEntityFeature.TARGET_TEMPERATURE_RANGE instead, please create a bug report at https://github.com/AlexxIT/SonoffLAN/issues

```
facing same issue 

also, energy sensor is unavailable
![image](https://github.com/AlexxIT/SonoffLAN/assets/83491212/21889d62-4af3-4528-b56b-98e3f67c22a5)

![image](https://github.com/AlexxIT/SonoffLAN/assets/83491212/c3f621c4-2e2d-43bf-b21e-41f7f91c1633)
updated to beta at 6:20 
![image](https://github.com/AlexxIT/SonoffLAN/assets/83491212/6105a4ca-cd0d-4de7-b61d-0b5581a42355)

**Comment by sambarlick** [comment]
I have the same issue, energy sensor no longer available on 2024.1.0b0.

**Comment by Traxeronus** [comment]
The same after 2024.1:

TEMP_CELSIUS was used from sonoff, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfTemperature.CELSIUS instead.

Logger: homeassistant.loader
Source: loader.py:822

Unexpected exception importing component custom_components.sonoff
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 822, in get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/config/custom_components/sonoff/__init__.py", line 17, in <module>
    from .core import backward, devices as core_devices
  File "/config/custom_components/sonoff/core/devices.py", line 16, in <module>
    from ..climate import XClimateTH, XClimateNS, XThermostat
  File "/config/custom_components/sonoff/climate.py", line 21, in <module>
    class XClimateTH(XEntity, ClimateEntity):
  File "/config/custom_components/sonoff/climate.py", line 27, in XClimateTH
    HVAC_MODE_OFF, HVAC_MODE_HEAT, HVAC_MODE_COOL, HVAC_MODE_DRY
    ^^^^^^^^^^^^^
NameError: name 'HVAC_MODE_OFF' is not defined

Logger: homeassistant.setup
Source: setup.py:251

Setup failed for custom integration 'sonoff': Unable to import component: Exception importing custom_components.sonoff
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 822, in get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/config/custom_components/sonoff/__init__.py", line 17, in <module>
    from .core import backward, devices as core_devices
  File "/config/custom_components/sonoff/core/devices.py", line 16, in <module>
    from ..climate import XClimateTH, XClimateNS, XThermostat
  File "/config/custom_components/sonoff/climate.py", line 21, in <module>
    class XClimateTH(XEntity, ClimateEntity):
  File "/config/custom_components/sonoff/climate.py", line 27, in XClimateTH
    HVAC_MODE_OFF, HVAC_MODE_HEAT, HVAC_MODE_COOL, HVAC_MODE_DRY
    ^^^^^^^^^^^^^
NameError: name 'HVAC_MODE_OFF' is not defined

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 251, in _async_setup_component
    component = integration.get_component()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 830, in get_component
    raise ImportError(f"Exception importing {self.pkg_path}") from err
ImportError: Exception importing custom_components.sonoff

**Comment by yousaf465** [comment]
https://discord.com/channels/330944238910963714/427516175237382144/1190699440193933456

https://github.com/AlexxIT/SonoffLAN/blob/56f20553d6a12daec99a888e5b01577591831c3b/custom_components/sonoff/core/entity.py#L40

```
You could also try removing that line manually and seeing if it fixes it.  If it breaks in a more spectacular way it’s likely means there is an inheritance problem in the construction of the entity in the integration

```
can anybody confirm if by removing this line it works?

**Comment by Traxeronus** [comment]
After commenting the line out the problem persists:

    # fix Hass v2021.12 empty attribute bug
    # _attr_is_on = None
    _attr_should_poll = False

**Comment by yousaf465** [comment]
> After commenting the line out the problem persists:
> 
> ```
> # fix Hass v2021.12 empty attribute bug
> # _attr_is_on = None
> _attr_should_poll = False
> ```

report in Discord too.

**Comment by AlexxIT** [comment]
https://github.com/AlexxIT/SonoffLAN/releases/tag/v3.5.4
